ads/kixer.js

```diff
@@ -43,7 +48,31 @@ export function kixer(global, data) {
       global.context.noContentAvailable();
     }
   };
-  d.addEventListener('load', kxload, false);
+  d.addEventListener('load', kxload, false); // Listen for the kixer load event
+
+  const kxviewCheck = function(intersectionEntry) {
+    inView = intersectionEntry.intersectionRatio > 0.5; // Half of the unit is in the viewport
+    if (inView === true) {
+      if (viewed === false && viewTimer == null) {
+        viewTimer = setTimeout(function() {
+          clearTimeout(viewTimer);
+          viewTimer = null;
+          if (inView === true) {
+            if (typeof __kx_viewability.process_locked === 'function') {
+              viewed = true;
+              __kx_viewability.process_locked(data.adslot);
+            }
+          }
+        }, 900);
+      }
+    }
+  };
```

Comment:
if you want it to be visible for 900ms before firing, should you reset the timer when `inView === false`?